cmd/argocd/commands/app.go

```diff
@@ -2588,8 +2600,8 @@ func waitOnApplicationStatus(ctx context.Context, acdClient argocdclient.Client,
 				fmt.Println("This is the state of the app after `wait` timed out:")
 			}
 
-			printFinalStatus(app)
 			cancel()
+			printFinalStatus(app)
```

Comment:
```suggestion
			printFinalStatus(app)
```
why are you printing after the `cancel()` has been called?

Reply:
Good point, thanks

Reply:
This is correct otherwise the ctx never gets called in the Get function. It is better to give a cancellation signal when the timeout has occured.